src/components/Header/defaultMenuLinks.js

```diff
@@ -29,23 +34,46 @@ export function defaultMenuLinks(
   const t = translate;
   const i18nOptions = { lng: locale };
 
-  return {
-    home: [{ label: 'TODO: defaultMenuLinks.js', href: '#TODO' }],
-    minimal: [
-      loginLink(t, deConsumer, primaryDomain),
-      localeLink(t, locale, subpath, switchLocaleLink),
-    ],
-    'logged-in': [
-      {
-        label: t('header.myApplicationsAndCoverage', i18nOptions),
-        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#landingPage`,
-      },
-      {
-        label: t('header.myProfile', i18nOptions),
-        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#settings`,
-      },
-    ],
-  };
+  // NOTE: order matters here and links will be displayed in order added to the arrays
+  const loggedOut = [];
+  const loggedIn = [
+    {
+      label: t('header.myApplicationsAndCoverage', i18nOptions),
+      href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#landingPage`,
+    },
+    {
+      label: t('header.myProfile', i18nOptions),
+      href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#settings`,
+    },
+  ];
+
+  if (!hideLanguageSwitch) {
+    const locLink = localeLink(t, locale, subpath, switchLocaleLink);
+    loggedOut.push(locLink);
+    loggedIn.push(locLink);
+  }
+
+  if (!hideLoginLink) {
+    const logLink = loginLink(t, deConsumer, primaryDomain);
+    loggedOut.push(Object.assign({ identifier: LINK_IDENTIFIERS.LOGIN }, logLink));
+  }
```

Comment:
[comment] So my biggest concern with moving the language, login, and logout buttons to this file is that if a product provides its own `links` prop to the header, then the language and logout buttons will not be added to the menu list by default. A product team would have to explicitly include those links in their custom list.

This leaves the potential that a product team may 1) forget to add these things, 2) position them in a way that is inconsistent with other products who are using the default links and 3) the product team will have to do some styling in their app to make the border line appear above a link like 'logout' to be consistent with the default link behavior.

Reply:
That's a good catch. I think I should be able to update the logic for the passed in links and append the ones that would be left out.